includes/class-wsuwp-shortcode-thankyoupage.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/**
+ * plugin class file for the idonate_thankyoupage shortcode
+ *
+ * Defines the functions necessary to register our custom thank you page enclosing shortcode
+ *
+ * @link       https://github.com/washingtonstateuniversity/WSUWP-Plugin-iDonate
+ * @since      0.0.20
+ *
+ * @package    WSUWP_Plugin_iDonate_ShortCode_ThankYouPage
+ * @author     Blair Lierman
+ */
+class WSUWP_Plugin_iDonate_ShortCode_ThankYouPage {
+
+	/**
+	 * Initializes the shortcode by registering the hooks necessary
+	 * for creating our shortcode within WordPress.
+	 *
+	 * @since 0.0.20
+	 */
+	public function init() {
+
+		add_shortcode( 'idonate_thankyoupage', array( $this, 'thankyoupage_create_shortcode' ) );
+	}
+
+	 /**
+	 * Defines the Thank You Page shortcode and applies the templating to the content
+	 * Usage: [idonate_thankyoupage]Thank You Page Content[/idonate_thankyoupage]
+	 *
+	 * @since 0.0.20
+	 */
+	public function thankyoupage_create_shortcode( $atts, $content = null ) {
+		$args = shortcode_atts( array(
+			'redirect_url' => get_site_url(),
+			'debug' => false,
+		), $atts );
+
+		$query_params = $this->parse_querystring();
+
+		if ( empty( $query_params['donor.id'] ) ) {
+			return 'Thank you for your donation. <br /> Please go to our <a href="' . esc_url( $args['redirect_url'] ) . '">Online Giving site</a> if you would like to give again.';
+		}
+
+		// Set a basic template if none given
+		if ( empty( $content ) ) {
+			$content = '
+			<h3><strong>Donation Summary:</strong></h3><br />
+			Donor number: {{donor.id}}<br />
+			Date: {{created}}<br />
+			Approval Code: {{transaction_id}}<br />
+
+			<p style="white-space: pre-line;">
+			<strong>Billing Information</strong>
+			{{donor.contact.firstname}} {{donor.contact.lastname}}
+			{{donor.contact.address.street}}
+			{{if donor.contact.address.street2}}{{donor.contact.address.street2}}{{end}}
+			{{donor.contact.address.city}}, {{donor.contact.address.state}} {{donor.contact.address.zip}}
+			{{donor.contact.address.country_name}}
+			{{donor.contact.phone}}
+			{{donor.contact.email}}
+			</p>
+
+			{{if comments}}Gift comments: {{comments}}{{end}}
+
+			{{if value}}
+			<p style="white-space: pre-line;">
+			Total Amount Charged: $<span />{{value}}
+			to your {{card_type}} with the last four {{last_four}}
+			</p>
+			{{end}}
+
+			<p>
+				<h5>WSU Foundation</h5><br />
+				<em>255 E. Main Street, Suite 201</em><br />
+				<em> PO Box 641927</em><br />
+				<em> Pullman, WA 99164-1927</em><br />
+				<em> Phone: 509-335-1686 or 800-GIV-2-WSU (448-2978)</em><br />
+				<em> Fax: 509-335-5903</em><br />
+				<em> E-mail: foundation@wsu.edu</em><br />
+			</p>
+			';
+		}
+
+		$content = $this->replace_conditionals( $content, $query_params );
+
+		foreach ( $query_params as $key => $value ) {
+			$content = str_replace( '{{' . $key . '}}', "$value", $content );
```

Comment:
It doesn't look like any of these params are escaped before being added to the content for output. As they all appear to be standard text data and don't require HTML tags, I'd replace with `esc_html( $value )`.